### PR TITLE
[merp] Include any managed methods in the 'unmanaged_frames' portion …

### PR DESCRIPTION
**Pull request by alexischr**
…of the output also

This can help correlate the two lists when we have mixed managed & unmanaged stacks in our crash output.

### CODE REVIEW

**Comment by alexischr** [comment]
@monojenkins build pkg

**Comment by leculver** [comment]
Would you mind changing protocol version from 0.0.4 to 0.0.5 for this change?  There's not a breaking change here per se but it will drastically change how we interpret these reports (for the better).

**Comment by alexischr** [comment]
@monojenkins build pkg

File: mono/mini/mini-exceptions.c [hunk 1]
```diff
@@ -1764,8 +1764,37 @@ mono_summarize_unmanaged_stack (MonoThreadSummary *out)
...
 		const char* module_buf = frame->unmanaged_data.module;
 		int success = mono_get_portable_ip (ip, &frame->unmanaged_data.ip, &frame->unmanaged_data.offset, &module_buf, (char *) frame->str_descr);
-		if (!success)
+
+		/* attempt to look up any managed method at that ip */
+		/* TODO: Trampolines - follow examples from mono_print_method_from_ip() */
+
+		MonoJitInfo *ji;
+		MonoDomain *domain = mono_domain_get ();
+		MonoDomain *target_domain = mono_domain_get ();
```
Comment (lambdageek) [inline mono/mini/mini-exceptions.c]: nit: don't need to initialize (or could initialize to NULL).  `mini_jit_info_table_find_ext` always overwrites `target_domain` .

File: mono/mini/mini-exceptions.c [hunk 2]
```diff
@@ -1764,8 +1764,37 @@ mono_summarize_unmanaged_stack (MonoThreadSummary *out)
...
+				MonoDotNetHeader *header = &image->image_info->cli_header;
+#ifndef MONO_PRIVATE_CRASHES
+				frame->managed_data.name = method->name;
+#endif
+				frame->managed_data.token = method->token;
+
+				frame->managed_data.guid = image->guid;
+				frame->managed_data.filename = image->filename;
+				frame->managed_data.image_size = header->nt.pe_image_size;
+				frame->managed_data.time_date_stamp = image->time_date_stamp;
```
Comment (lambdageek) [inline mono/mini/mini-exceptions.c]: Could you make a helper function to share code that's filling in `managed_data` here with the code in `summarize_frame_internal` that's doing the same thing.

File: mono/mini/mini-exceptions.c [hunk 3]
```diff
@@ -1764,8 +1774,36 @@ mono_summarize_unmanaged_stack (MonoThreadSummary *out)
...
+				MonoMethod *method = jinfo_get_method (ji);
+				MonoImage *image = mono_class_get_image (method->klass);
+				MonoDotNetHeader *header = &image->image_info->cli_header;
+#ifndef MONO_PRIVATE_CRASHES
+				frame->managed_data.name = method->name;
+#endif
+				frame->managed_data.token = method->token;
+
+				frame->managed_data.guid = image->guid;
+				frame->managed_data.filename = basename_ptr(image->filename);
```
Comment (lambdageek) [inline mono/mini/mini-exceptions.c]: Why is this different from `summarize_frame_internal` (which uses `image->module_name` here)

Comment (alexischr) [reply mono/mini/mini-exceptions.c]: Typo; fixed

**Comment by lambdageek** [comment]
@monojenkins build failed

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-10
@monojenkins backport to 2019-08

**Comment by lambdageek** [comment]
Ah. only 2019-08 needed the backport.  2019-10 already includes 10e9159eaa92364f03c8359c47e6129aba1f9bda because we branched after it was merged.

**Pull request merged**